lib/closure_tree/support.rb

```diff
@@ -32,7 +32,12 @@ def initialize(model_class, options)
     end
 
     def hierarchy_class_for_model
-      hierarchy_class = model_class.parent.const_set(short_hierarchy_class_name, Class.new(ActiveRecord::Base))
+      parent_class = if ActiveSupport.version >= Gem::Version.new('6.0.0.beta1')
```

Comment:
may want to `to_s` for the comparison - it simplifies things

Also, rails 6.0 has been released, so probably want to fix that too